dds/src/Subscriber.cpp

```diff
@@ -36,20 +36,26 @@ namespace dds {
 Subscriber::Subscriber(
         Participant* participant,
         const std::string& topic_name,
-        const std::string& message_type,
+        const ::xtypes::DynamicType& message_type,
         TopicSubscriberSystem::SubscriptionCallback soss_callback)
 
     : topic_name_{topic_name}
     , message_type_{message_type}
     , soss_callback_{soss_callback}
     , reception_threads_{}
 {
-    dynamic_data_ = participant->create_dynamic_data(message_type);
+    DynamicTypeBuilder* builder = Conversion::create_builder(message_type);
+    if (builder != nullptr)

```

Comment:
It can be null if the type contains unsupported types, which currently are none (in the other way, from dss to soss, it is possible).
Added a throw, just in case.

Reply:
ok, :+1: